dev/archery/archery/integration/tester_cpp.py

```diff
@@ -0,0 +1,105 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import contextlib
+import os
+import subprocess
+
+from .tester import Tester
+from .util import run_cmd, ARROW_ROOT_DEFAULT
+
+
+class CPPTester(Tester):
+    PRODUCER = True
+    CONSUMER = True
+    FLIGHT_SERVER = True
+    FLIGHT_CLIENT = True
+
+    EXE_PATH = os.environ.get(
```

Comment:
You might want to pass a [CMakeBuild](https://github.com/apache/arrow/blob/master/dev/archery/archery/utils/cmake.py#L120) instance here, it accomplishes what you want with `CMakeBuild.binaries_dir()`. You could extend and implement `CMakeBuild.binary(exe_name)` which would return a proper instance of util.Command with path taken care of.